src/energy_systems/heat_sources/generic_heat_source.jl

```diff
@@ -0,0 +1,201 @@
+"""
+A generic heat source that models a number of different technologies used to supply heat.
+
+Typically these are sources of low temperature heat for heat pumps, that only supply heat
+and do not store it. This makes them different from other sources e.g. geothermal sources.
+Examples of technologies: river water, lake water, waste water, atmosphere, industrial
+processes or one-way connections to a heat network.
+"""
+mutable struct GenericHeatSource <: Component
+    uac::String
+    controller::Controller
+    sys_function::SystemFunction
+    medium::Symbol
+
+    input_interfaces::InterfaceMap
+    output_interfaces::InterfaceMap
+
+    max_power_profile::Union{Profile,Nothing}
+    temperature_profile::Union{Profile,Nothing}
+    scaling_factor::Float64
+    constant_power::Union{Nothing,Float64}
+    constant_temperature::Temperature
+
+    temperature_reduction_model::String
+    min_source_in_temperature::Temperature
+    max_source_in_temperature::Temperature
+    avg_source_in_temperature::Temperature
+    lmtd_min::Float64
+
+    max_energy::Float64
+    temperature_src_in::Temperature
+    temperature_snk_out::Temperature
+
+    function GenericHeatSource(
+        uac::String,
+        config::Dict{String,Any},
+        sim_params::Dict{String,Any}
+    )
+        max_power_profile = "max_power_profile_file_path" in keys(config) ?
+            Profile(config["max_power_profile_file_path"], sim_params) :
+            nothing
+
+        temperature_profile = get_temperature_profile_from_config(config, sim_params, uac)
+
+        medium = Symbol(config["medium"])
+        register_media([medium])
+
+        return new(
+            uac, # uac
+            controller_for_strategy( # controller
+                config["strategy"]["name"], config["strategy"], sim_params
+            ),
+            sf_bounded_source, # sys_function
+            medium, # medium
+            InterfaceMap( # input_interfaces
+                medium => nothing
+            ),
+            InterfaceMap( # output_interfaces
+                medium => nothing
+            ),
+            max_power_profile,
+            temperature_profile,
+            default(config, "scale", 1.0), # scaling_factor
+            default(config, "constant_power", nothing),
+            default(config, "constant_temperature", nothing),
+            default(config, "temperature_reduction_model", "none"),
```

Comment:
Should we add a check of the input values of temperature_reduction_model in the constructor to keep it in mind for the future check_inputs?